Error when installing on MacOS

I get the following error when installing `numba-kdtree` on MacOS using `python3 -m pip install numba-kdtree`:

```
Collecting numba-kdtree
  Using cached numba-kdtree-0.1.7.tar.gz (25 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: numba>=0.52 in ./opt/miniconda3/lib/python3.9/site-packages (from numba-kdtree) (0.55.1)
Requirement already satisfied: setuptools in ./opt/miniconda3/lib/python3.9/site-packages (from numba>=0.52->numba-kdtree) (58.0.4)
Requirement already satisfied: llvmlite<0.39,>=0.38.0rc1 in ./opt/miniconda3/lib/python3.9/site-packages (from numba>=0.52->numba-kdtree) (0.38.0)
Requirement already satisfied: numpy<1.22,>=1.18 in ./opt/miniconda3/lib/python3.9/site-packages (from numba>=0.52->numba-kdtree) (1.21.6)
Building wheels for collected packages: numba-kdtree
  Building wheel for numba-kdtree (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/josafat/opt/miniconda3/bin/python3 /Users/josafat/opt/miniconda3/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /var/folders/pk/16kqwmls23b61lxyzf865kx00000gn/T/tmpc68c4xhx
       cwd: /private/var/folders/pk/16kqwmls23b61lxyzf865kx00000gn/T/pip-install-9ait71fi/numba-kdtree_16be559bf92242839f3434960bf36a3b
  Complete output (47 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-cpython-39
  creating build/lib.macosx-10.9-x86_64-cpython-39/numba_kdtree
  copying numba_kdtree/ckdtree_ctypes.py -> build/lib.macosx-10.9-x86_64-cpython-39/numba_kdtree
  copying numba_kdtree/_version.py -> build/lib.macosx-10.9-x86_64-cpython-39/numba_kdtree
  copying numba_kdtree/__init__.py -> build/lib.macosx-10.9-x86_64-cpython-39/numba_kdtree
  copying numba_kdtree/kd_tree.py -> build/lib.macosx-10.9-x86_64-cpython-39/numba_kdtree
  running build_ext
  building 'numba_kdtree._ckdtree' extension
  creating build/temp.macosx-10.9-x86_64-cpython-39
  creating build/temp.macosx-10.9-x86_64-cpython-39/src
  creating build/temp.macosx-10.9-x86_64-cpython-39/src/ckdtree
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/josafat/opt/miniconda3/include -arch x86_64 -I/Users/josafat/opt/miniconda3/include -fPIC -O2 -isystem /Users/josafat/opt/miniconda3/include -arch x86_64 -I/Users/josafat/opt/miniconda3/include/python3.9 -I/private/var/folders/pk/16kqwmls23b61lxyzf865kx00000gn/T/pip-install-9ait71fi/src/ckdtree -I/Users/josafat/opt/miniconda3/include/python3.9 -c src/ckdtree/build.cpp -o build/temp.macosx-10.9-x86_64-cpython-39/src/ckdtree/build.o -fPIC -O3 -DKDTREE_COMPILING=1
  In file included from src/ckdtree/build.cpp:10:
  src/ckdtree/ckdtree_decl.h:61:19: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
       ~ckdtree() = default;
                    ^
  src/ckdtree/ckdtree_decl.h:64:30: error: a space is required between consecutive right angle brackets (use '> >')
      std::vector<ckdtreenode<T>> tree_buffer;
                               ^~
                               > >
  src/ckdtree/build.cpp:22:22: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
      using DataType = T;
                       ^
  src/ckdtree/build.cpp:23:25: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
      using ckdtreenode = ckdtreenode<DataType>;
                          ^
  src/ckdtree/build.cpp:30:5: warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
      auto *indices = (ckdtree_intp_t *)(self->raw_indices);
      ^
  src/ckdtree/build.cpp:37:23: error: no member named 'emplace_back' in 'std::vector<ckdtreenode<float> >'
      self->tree_buffer.emplace_back(start_idx, end_idx);
      ~~~~~~~~~~~~~~~~~ ^
  src/ckdtree/build.cpp:201:5: note: in instantiation of function template specialization 'build<float>' requested here
      build<float>(self, start_idx, end_idx, mins, maxes, _balanced, _compact);
      ^
  src/ckdtree/build.cpp:37:23: error: no member named 'emplace_back' in 'std::vector<ckdtreenode<double> >'
      self->tree_buffer.emplace_back(start_idx, end_idx);
      ~~~~~~~~~~~~~~~~~ ^
  src/ckdtree/build.cpp:212:5: note: in instantiation of function template specialization 'build<double>' requested here
      build<double>(self, start_idx, end_idx, mins, maxes, _balanced, _compact);
      ^
  4 warnings and 3 errors generated.
  error: command '/usr/bin/clang' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for numba-kdtree
Failed to build numba-kdtree
ERROR: Could not build wheels for numba-kdtree which use PEP 517 and cannot be installed directly
```
This is because the `clang++` compiler on my system uses the C++98 standard per default (even though `man clang++` says `The default C++ language standard is gnu++14.`). I was able to fix this by adding the `-std=c++11` flag to the compilation arguments in the `setup.py`.

--- a/setup.py
+++ b/setup.py
@@ -73,6 +73,8 @@ if platform.system() == "Windows":
     extra_compile_args = ['/O2', '/DKDTREE_COMPILING=1']
 else:
     extra_compile_args = ['-fPIC', '-O3', '-DKDTREE_COMPILING=1']
+    if platform.system() == "Darwin":
+        extra_compile_args.append('-std=c++11')
 
 module = CTypesExtension('numba_kdtree._ckdtree',
                    sources=ckdtree_src,

